src/layout/IHyprLayout.cpp

```diff
@@ -477,18 +475,28 @@ void IHyprLayout::changeWindowFloatingMode(CWindow* pWindow) {
 
         g_pCompositor->changeWindowZOrder(pWindow, true);
 
-        if (DELTALESSTHAN(pWindow->m_vRealSize.vec().x, pWindow->m_vLastFloatingSize.x, 10) && DELTALESSTHAN(pWindow->m_vRealSize.vec().y, pWindow->m_vLastFloatingSize.y, 10)) {
-            pWindow->m_vRealPosition = pWindow->m_vRealPosition.goalv() + (pWindow->m_vRealSize.goalv() - pWindow->m_vLastFloatingSize) / 2.f + Vector2D{10, 10};
-            pWindow->m_vRealSize     = pWindow->m_vLastFloatingSize - Vector2D{20, 20};
+        const auto PSAVEDSIZE = pWindow->m_vRealSize.goalv();
+
+        Vector2D   sizeDelta     = {0, 0};
```

Comment:
uhh... why?

Reply:
[vaxerski](https://github.com/vaxerski) We try to use lastfloatposition whenever possible if it has been saved, and change it only if position.x == 0 or position.y == 0, so we add a delta size and position for windows that themselves do not know how to switch to float(allways maximazed ), so we crop the window correctly, I increased the window crop size. If we do not use lastfloatposition, then there are situations when the window after toggle float loses its coordinates and begins to be drawn in a different place.

Reply:
@vaxerski Also in some checks for == 0, I replaced it with <= 0 because In some places the coordinates and dimensions may be negative.

Reply:
mhm, 50 too much though, maybe 20?

Reply:
@vaxerski So now there are no small screens like 480x320, and smartphones also have larger screens. On a 480x320 screen, the window after cropping will have a size of 380x220, which visually makes the window 1/3 smaller. 100 seemed optimal to me to see that the window became float.

Reply:
fair enough